cub/cub/device/device_reduce.cuh

```diff
@@ -347,6 +348,35 @@ struct DeviceReduce
   }
 #endif // DOXYGEN_SHOULD_SKIP_THIS
 
+  template <typename InputIteratorT, typename OutputIteratorT, typename NumItemsT>
+  CUB_RUNTIME_FUNCTION static cudaError_t DeterministicSum(
+    void* d_temp_storage,
+    size_t& temp_storage_bytes,
+    InputIteratorT d_in,
+    OutputIteratorT d_out,
+    NumItemsT num_items,
+    cudaStream_t stream = 0)
```

Comment:
important: we haven't converged on API yet. Let's remove this function from public API for now and move it to tests.